backend/models/databases/supabase/brains.py

```diff
@@ -0,0 +1,192 @@
+            .execute()
+        )
+
+        return response.data
+
+    def create_brain_vector(self, brain_id, vector_id, file_sha1):
+        response = (
+            self.db.table("brains_vectors")
+            .insert(
+                {
+                    "brain_id": str(brain_id),
+                    "vector_id": str(vector_id),
+                    "file_sha1": file_sha1,
+                }
+            )
+            .execute()
+        )
+        return response.data
+
+    def get_vector_ids_from_file_sha1(self, file_sha1: str):
+        # move to vectors class
+        vectorsResponse = (
+            self.db.table("vectors")
+            .select("id")
+            .filter("metadata->>file_sha1", "eq", file_sha1)
+            .execute()
+        )
+        return vectorsResponse.data
+
+    def update_brain_fields(self, brain_id, brain_name):
+        self.db.table("brains").update({"name": brain_name}).match(
+            {"brain_id": brain_id}
+        ).execute()
+
+    def get_brain_vector_ids(self, brain_id):
+        """
+        Retrieve unique brain data (i.e. uploaded files and crawled websites).
+        """
+
+        response = (
+            self.db.from_("brains_vectors")
+            .select("vector_id")
+            .filter("brain_id", "eq", brain_id)
+            .execute()
+        )
+
+        vector_ids = [item["vector_id"] for item in response.data]
+
+        if len(vector_ids) == 0:
+            return []
+
+        return vector_ids
+
+    def delete_file_from_brain(self, brain_id, file_name: str):
+        # First, get the vector_ids associated with the file_name
+        vector_response = (
+            self.db.table("vectors")
+            .select("id")
+            .filter("metadata->>file_name", "eq", file_name)
+            .execute()
+        )
+        vector_ids = [item["id"] for item in vector_response.data]
+
+        # For each vector_id, delete the corresponding entry from the 'brains_vectors' table
+        for vector_id in vector_ids:
+            self.db.table("brains_vectors").delete().filter(
+                "vector_id", "eq", vector_id
+            ).filter("brain_id", "eq", brain_id).execute()
+
+            # Check if the vector is still associated with any other brains
+            associated_brains_response = (
+                self.db.table("brains_vectors")
+                .select("brain_id")
+                .filter("vector_id", "eq", vector_id)
+                .execute()
+            )
+            associated_brains = [
+                item["brain_id"] for item in associated_brains_response.data
+            ]
+
+            # If the vector is not associated with any other brains, delete it from 'vectors' table
+            if not associated_brains:
+                self.db.table("vectors").delete().filter(
+                    "id", "eq", vector_id
+                ).execute()
+
+        return {"message": f"File {file_name} in brain {brain_id} has been deleted."}
```

Comment:
Let's change this logic in another MR to instead have the methods of the repo not returning objects instead returning true/false or errors? 
The message here feels like app logic and probably should not be in the repository. 

The message is also wrong when the method is called on a filename/braid_id that does not exist

Reply:
I'll change this. I missed it.